atlas/core/imageManager.cpp

```diff
@@ -48,94 +48,95 @@ namespace imageManager
-			const auto dest_root { config::paths::images::getPath() / std::to_string( game_id ) };
-			std::filesystem::create_directories( dest_root );
-
-			const auto hashData = []( const char* data_ptr, const int size ) -> QByteArray
-			{
-				ZoneScopedN( "Hash" );
-				QCryptographicHash hash { QCryptographicHash::Sha256 };
+			atlas::logging::userwarn( fmt::format(
+				"importImage failed. Attempted to open file {} which doesn't exist anymore. Wrong permissions?",
+				path ) );
+			throw std::runtime_error( fmt::format( "Filepath {} does not exist. Unable to add as image", path ) );
+		}
 
-				hash.addData( { data_ptr, size } );
+		//Load file so we have direct access to the bytearray
+		QFile file( path );
+		if ( !file.open( QFile::ReadOnly ) )
+			throw std::runtime_error( fmt::format( "Failed to load image from file: {}", path ) );
+		TracyCZoneN( tracy_ImageLoad, "Image load", true );
+		const QByteArray byteArray { file.readAll() };
+		TracyCZoneEnd( tracy_ImageLoad );
+		file.close();
+
+		//Store image inside of QImage
+		QImage temp_image;
+		const bool load_success { temp_image.loadFromData( byteArray ) };
+		if ( !load_success )
+			throw std::runtime_error( fmt::format( "Failed to store image from byte array: {}", path.filename() ) );
+
+		const std::string ext { path.extension().string().substr( 1 ) };
+		const auto dest_root { config::paths::images::getPath() / std::to_string( game_id ) };
+		std::filesystem::create_directories( dest_root );
+
+		const auto hashData = []( const char* data_ptr, const int size ) -> QByteArray
+		{
+			ZoneScopedN( "Hash" );
+			QCryptographicHash hash { QCryptographicHash::Sha256 };
 
-				return hash.result();
-			};
+			hash.addData( { data_ptr, size } );
 
-			//If GIF then store, do not convert
-			if ( ext == "gif" )
+			return hash.result();
+		};
+
+		//If GIF then store, do not convert
+		if ( ext == "gif" )
+		{
+			const auto hash { hashData( byteArray, static_cast< int >( byteArray.size() ) ) };
+			const auto dest { dest_root / ( hash.toHex().toStdString() + path.extension().string() ) };
+
+			//Qt is stupid and will not save gifs...  so we have to copy it
+			const bool file_copied { std::filesystem::copy_file( path, dest ) };
+			if ( !file_copied )
+				throw std::runtime_error( fmt::format( "Unable to save gif to images folder: {}", path.filename() ) );
+
+			return dest;
+		}
+		else
+		{
+			const std::string image_type { config::images::image_type::get().toStdString() };
+
+			TracyCZoneN( tracy_SaveImage, "Image save to buffer as WEBP", true );
+			QByteArray webp_byteArray;
+			QBuffer webp_buffer( &webp_byteArray );
+			temp_image.save( &webp_buffer, image_type.c_str(), 90 );
+			TracyCZoneEnd( tracy_SaveImage );
+
+			//Which is bigger?
+			if
+			(
+				( webp_buffer.size() >= byteArray.size() )	// Is WebP bigger?
+				|| ( temp_image.width() > webp_max )		// Is it too wide for WebP?
+				|| ( temp_image.height() > webp_max )		// Is it too tall for WebP?
```

Comment:
Can this check be moved up at all? `temp_image.save()` is a pretty expensive process so if we do an early check we can avoid it when webp can't support the image size (Very very VERY unlikely but still preferred to be pessimistic)